File: Day-10/LongestBinarySubSequence.go
package main

func longestSubsequence(s string, k int) int {
	length := 0
	value := 0

	for i := len(s) - 1; i >= 0; i-- {
			if s[i] == '0' {
					length++
			} else {
					if length < 30 && (value|(1<<length)) <= k {
							value |= (1 << length)
							length++
					}
			}
	}

	return length
}

File: Day-10/LongestBinarySubSequence.py
def longestSubsequence(s: str, k: int) -> int:
    length = 0
    value = 0

    for i in reversed(range(len(s))):
        if s[i] == '0':
            length += 1
        else:
            if length < 30 and (value | (1 << length)) <= k:
                value |= (1 << length)
                length += 1

    return length